### PR TITLE
New actionable proposals services

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Add `updateActionableProposals` to fill the `actionableNnsProposalsStore`.

# Changes

- `updateActionableProposals` service

# Tests

- `updateActionableProposals` service

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary

### CODE REVIEW

File: frontend/src/lib/services/actionable-proposals.services.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  votableNeurons,
+  type NeuronInfo,
+} from "@dfinity/nns";
+import { isNullish, nonNullish } from "@dfinity/utils";
+import { get } from "svelte/store";
+
+/**
+ * Fetch all proposals that are accepting votes and put them in the store.
+ */
+export const updateActionableProposals = async (): Promise<void> => {
```
Comment (dskloetd) [inline frontend/src/lib/services/actionable-proposals.services.ts:19]: I've seen functions like this, usually been called `load...`, I think.
Or does that mean something else?

File: frontend/src/tests/lib/services/actionable-proposals.services.spec.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      );
+    });
+
+    it("should update actionable nns proposals store with votable proposals only", async () => {
+      expect(spyQueryProposals).not.toHaveBeenCalled();
+
+      await updateActionableProposals();
+
+      const { proposals } = get(actionableNnsProposalsStore);
+      expect(proposals).toHaveLength(1);
```
Comment (dskloetd) [inline frontend/src/tests/lib/services/actionable-proposals.services.spec.ts:97]: This is redundant.

File: frontend/src/tests/lib/services/actionable-proposals.services.spec.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            topics: [],
+          },
+        })
+      );
+    });
+
+    it("should update actionable nns proposals store with votable proposals only", async () => {
+      expect(spyQueryProposals).not.toHaveBeenCalled();
+
+      await updateActionableProposals();
```
Comment (dskloetd) [inline frontend/src/tests/lib/services/actionable-proposals.services.spec.ts:94]: Can you check that the store is empty before this call?

File: frontend/src/tests/lib/services/actionable-proposals.services.spec.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+      const { proposals } = get(actionableNnsProposalsStore);
+      expect(proposals).toHaveLength(1);
+      expect(proposals).toEqual([votableProposal]);
+    });
+  });
```
Comment (dskloetd) [inline frontend/src/tests/lib/services/actionable-proposals.services.spec.ts:100]: Can you add:
1. A test that no queries are made if the store is already filled.
2. A test that the neurons store is filled.

File: frontend/src/lib/services/actionable-proposals.services.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/services/actionable-proposals.services.ts]: True, thanks, updated.

File: frontend/src/tests/lib/services/actionable-proposals.services.spec.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/services/actionable-proposals.services.spec.ts]: Agree.

File: frontend/src/tests/lib/services/actionable-proposals.services.spec.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/services/actionable-proposals.services.spec.ts]: Done.

File: frontend/src/tests/lib/services/actionable-proposals.services.spec.ts (hunk 4 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/services/actionable-proposals.services.spec.ts:130]: 1. Good point, added.
2. done - https://github.com/dfinity/nns-dapp/pull/4543/files#diff-0bf8b9e0ec74a580a90f62a13f8eff5eaffde5a1c9652dded581b719c5926fe6R73

**Review by dskloetd — APPROVED** [review]
Thanks!

**Pull request merged**